canonical-webteam/canonicalwebteam.docs
# Create extension

# Summary

Fixes https://github.com/ubuntudesign/base-squad/issues/450

First extension for discourse based docs.
This follows this flask extension pattern: http://flask.pocoo.org/docs/1.0/extensiondev/

# QA

In a python 3 environment:
- create directory: `mkdir flaskapp`
- `cd flaskapp`
- `vim requirements.txt`
```
Flask
-e git+https://github.com/tbille/canonicalwebteam.docs.git@create-extention#egg=canonicalwebteam.docs
```

- `vim app.py`
```
import flask
from canonicalwebteam.discourse_docs import DiscourseDocs

app = flask.Flask(__name__)

discourse = DiscourseDocs()
discourse.init_app(app, "/", "https://forum.snapcraft.io/", 3781, 15)
```

- `mkdir templates`
- `mkdir templates/docs`
- in this directory created move those files from this repo: https://github.com/canonical-websites/docs.snapcraft.io/tree/master/templates

(You can only move `document.html`  to start, just remove the block content and the layout)

- `pip install -r requirements.txt`
- `FLASK_DEBUG=True FLASK_APP=app:app FLASK_ENV=development flask run`
- http://127.0.0.1:5000/
- You should have the documentation (without styling)
- Try to play around to see if all the functionnalities are working

# Needs to be done

- [x] run tests in circle ci
- [ ] extract hardcoded templates directory to render templates
